lib/statem_dsl.ex

```diff
@@ -120,24 +115,33 @@ defmodule PropCheck.StateM.DSL do
   ## Additional model elements
 
   In addition to commands, we need to define the model itself. This is the
-  ingenious part of stateful property based testing! The initial state
-  of the model must be implemented as the function `initial_state/0`. From this
+  ingenious part of stateful property based testing! The initial state of the
+  model must be implemented as the function `initial_state/0`. It doesn't accept
+  any arguments, because this function has to be deterministic. From this
   function, all model evolutions start. In our simplified cache example the
   initial model is an empty list:
 
       def initial_state(), do: []
 
-  The commands are generated with the same frequency by default. Often, this
-  is not appropriate, e.g. in the cache example we expect many more `find` than
-  `cache` commands. Therefore, commands can have a weight, which is technically used
-  inside a `PropCheck.BasicTypes.frequency/1` generator. The weights are defined
-  in callback function `c:weight/1`, taking the current model state and returning
-  a map of command and frequency pairs to be generated.  In our cache example
-  we want the `find` command to appear three times more often than other commands:
-
-      def weight(_state), do: %{find: 3, cache: 1, flush: 1}
+  Sequence of commands to be run is generated repeatedly by `c:command_gen/1`
+  callback. The generator has to return a tuple of with a command name and a
+  list of it's arguments (a list of generators). This callback expects current
+  state as an argument, which often is used to determine next one from a set of
+  appropriate commands (e.g. there might not be much sens in `delete_user`
+  command, if there are no users in the system yet). Usually a
```

Comment:
language: This callback expects *the* current  state as an argument, which often is used to determine *the* next one from a set of  appropriate commands (e.g. there might not be much *sense* in *calling the *`delete_user` command, if ...)

Reply:
Right!